microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,81 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+from typing import Dict
+
+from microcosm_eventsource.factory import EventFactory
+
+
+container_common_mutator_registry: Dict[str, list] = defaultdict(list)
+container_event_specific_mutator_registry: Dict[str, list] = defaultdict(list)
+
+
+def register_container_mutator_common(event_type):
+    def decorator(func):
+        container_common_mutator_registry[event_type.__name__].append(func.__name__)
```

Comment:
I am not convinced supporting a chain of update functions (which could potentially be order dependent) is such a good idea. I'd almost prefer to have the convention be at most ONE update will happen. Either a base update for all events or an event specific update. I think this will be more explicit/clear otherwise the order of `mutators` (or `aggregators` as I call them later on) might cause weird consequences.

If we don't support a nice chaining of `aggregators`, I realize we may duplicate some logic, but I think it will lead to more explicit code and generally less pain.

Reply:
The reason for me to go down the route of multiple update functions, was such that there maybe events which have overlapping update logic. Though, it is fair to be more explicit, and I am open to changing this.